telebot/modules/delete.py

```diff
@@ -0,0 +1,144 @@
+from functools import wraps
+from re import match
+from traceback import print_exc
+from typing import Callable
+
+from telegram import Update
+from telegram.error import BadRequest
+from telegram.ext import CallbackContext, CommandHandler
+
+from telebot import dispatcher
+from telebot.functions import (
+    check_user_admin,
+    check_bot_admin,
+    bot_action,
+    check_reply_to_message,
+)
+from telebot.modules.db.exceptions import get_command_exception_chats
+from telebot.modules.db.users import add_user
+
+
+def check_bot_can_delete(func: Callable):
+    """
+    Wrapper function for checking if bot can delete messages
+    :param func: The function this wraps over
+    """
+
+    @wraps(func)
+    def wrapper(update: Update, context: CallbackContext, *args, **kwargs):
+        # check if bot has perms to delete a message
+        if (
+            update.effective_chat.type != "private"
+            and not update.effective_chat.get_member(context.bot.id).can_delete_messages
+        ):
+            update.effective_message.reply_text(
+                "Bribe your sugar daddy with some catnip and ask him to allow me to delete messages..."
+            )
+            return
+
+        return func(update, context, *args, **kwargs)
+
+    return wrapper
+
+
+def check_user_can_delete(func: Callable):
+    """
+    Wrapper function for checking if user can delete messages
+    :param func: The function this wraps over
+    """
+
+    @wraps(func)
+    def wrapper(update: Update, context: CallbackContext, *args, **kwargs):
+        # check if user can delete messages
+        if (
+            update.effective_chat.type != "private"
+            and update.effective_message.reply_to_message.from_user.id != update.effective_user.id
+            and update.effective_chat.id not in get_command_exception_chats("delete")
+        ):
+            user = update.effective_chat.get_member(update.effective_user.id)
+            if not user.can_delete_messages and user.status != "creator":
```

Comment:
Creator will always have delete messages permission nah?

Reply:
well, I remember fetching the perms of myself on a group I created and I had no perms..... apparently the creator status is a big bypass